lib/internal/modules/package_json_reader.js

```diff
@@ -154,10 +190,145 @@ function getPackageType(url) {
+    specifier : StringPrototypeSlice(specifier, 0, separatorIndex);
+
+  // Package name cannot have leading . and cannot have percent-encoding or
+  // \\ separators.
+  if (RegExpPrototypeExec(invalidPackageNameRegEx, packageName) !== null) {
+    validPackageName = false;
+  }
+
+  if (!validPackageName) {
+    throw new ERR_INVALID_MODULE_SPECIFIER(
+      specifier, 'is not a valid package name', fileURLToPath(base));
+  }
+
+  const packageSubpath = '.' + (separatorIndex === -1 ? '' :
+    StringPrototypeSlice(specifier, separatorIndex));
+
+  return { packageName, packageSubpath, isScoped };
+}
+
+function getPackageJSONURL(specifier, base) {
+  const { packageName, packageSubpath, isScoped } =
+    parsePackageName(specifier, base);
+
+  // ResolveSelf
+  const packageConfig = getPackageScopeConfig(base);
+  if (packageConfig.exists) {
+    if (packageConfig.exports != null && packageConfig.name === packageName) {
+      const packageJSONPath = packageConfig.pjsonPath;
+      return { packageJSONUrl: pathToFileURL(packageJSONPath), packageJSONPath, packageSubpath };
+    }
+  }
+
+  let packageJSONUrl =
+    new URL('./node_modules/' + packageName + '/package.json', base);
+  let packageJSONPath = fileURLToPath(packageJSONUrl);
+  let lastPath;
+  do {
+    const stat = internalFsBinding.internalModuleStat(
+      internalFsBinding,
+      StringPrototypeSlice(packageJSONPath, 0, packageJSONPath.length - 13),
+    );
+    // Check for !stat.isDirectory()
+    if (stat !== 1) {
+      lastPath = packageJSONPath;
+      packageJSONUrl = new URL((isScoped ?
+        '../../../../node_modules/' : '../../../node_modules/') +
+        packageName + '/package.json', packageJSONUrl);
+      packageJSONPath = fileURLToPath(packageJSONUrl);
+      continue;
+    }
+
+    // Package match.
+    return { packageJSONUrl, packageJSONPath, packageSubpath };
+  } while (packageJSONPath.length !== lastPath.length);
+
+  throw new ERR_MODULE_NOT_FOUND(packageName, fileURLToPath(base), null);
+}
+
+const pjsonImportAttributes = { __proto__: null, type: 'json' };
+let cascadedLoader;
+/**
+ * @param {URL['href'] | string | URL} specifier The location for which to get the "root" package.json
+ * @param {URL['href'] | string | URL} [base] The location of the current module (ex file://tmp/foo.js).
+ */
+function findPackageJSON(specifier, base = 'data:') {
+  if (arguments.length === 0) {
+    throw new ERR_MISSING_ARGS('specifier');
+  }
+  try {
+    specifier = `${specifier}`;
+  } catch {
+    validateString(specifier, 'specifier');
+  }
+
+  let parentURL;
+  if (isURL(base)) {
+    parentURL = new URL(base);
+  } else {
+    validateString(base, 'base');
+    if (
+      path.isAbsolute(base) ||
+      (URLCanParse(base) && !StringPrototypeStartsWith(base, 'file:'))
+    ) {
+      parentURL = pathToFileURL(path.toNamespacedPath(base));
+    } else {
+      parentURL = URL.parse(base) || pathToFileURL(base);
+    }
```

Comment:
```suggestion
  if (!isURL(base) {
    validateString(base, 'base');
    if (path.isAbsolute(base)) {
      parentURL = pathToFileURL(base);
    } else {
      parentURL = new URL(base);
    }
```

Reply:
That does not work. I already tried it.

Reply:
What does not work? I've tried it locally and didn't get any test failure

Reply:
On windows. I created a test locally and simulates the windows issue, and it continued to fail with this. It can't fully pass on mac because various node path tools behave differently so I couldn't truly simulate, but it can get further the relevant issue (which means it will succeed on windows).